lib/top/mdx_top.ml

```diff
@@ -796,6 +796,7 @@ let rec save_summary acc s =
 #endif
 #if OCAML_MAJOR >= 4 && OCAML_MINOR >= 8
     | Env_persistent (summary, _)
+    | Env_module (summary, _, _, _)
```

Comment:
I would move that line up to keep all the different handlings on `Env_module` together. In particular, this one looks wrong: you're not collecting the ident.